src/libsync/theme.cpp

```diff
@@ -899,16 +902,43 @@ QColor Theme::errorBoxBorderColor() const
     return QColor{"black"};
 }
 
-QPalette Theme::systemPalette()
+void Theme::connectToPaletteSignal()
 {
     if(!_guiAppInstance) {
         const auto ptr = qobject_cast<QGuiApplication *>(QGuiApplication::instance());
         if(ptr) {
             _guiAppInstance.reset(ptr);
-            connect(ptr, &QGuiApplication::paletteChanged, this, &Theme::systemPaletteChanged);
+            connect(_guiAppInstance.data(), &QGuiApplication::paletteChanged, this, &Theme::systemPaletteChanged);
+            connect(_guiAppInstance.data(), &QGuiApplication::paletteChanged, this, &Theme::darkModeChanged);
         }
     }
+}
+
+QPalette Theme::systemPalette()
+{
+    connectToPaletteSignal();
+#if defined(Q_OS_WIN)
+    if(darkMode()) {
+        return reserveDarkPalette;
+    }
+#endif
     return QGuiApplication::palette();
 }
 
+bool Theme::darkMode()
+{
+    connectToPaletteSignal();
+// Windows: Check registry for dark mode
+#if defined(Q_OS_WIN)
+    const auto darkModeSubkey = QStringLiteral("Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize");
+    if (!Utility::registryKeyExists(HKEY_CURRENT_USER, darkModeSubkey)) {
+        return false;
+    }
+    const auto darkMode = !Utility::registryGetKeyValue(HKEY_CURRENT_USER, darkModeSubkey, QStringLiteral("AppsUseLightTheme")).toBool();
```

Comment:
@claucambra I'd try and combine `if (!Utility::registryKeyExists(HKEY_CURRENT_USER, darkModeSubkey))` and `!Utility::registryGetKeyValue(HKEY_CURRENT_USER, darkModeSubkey, QStringLiteral("AppsUseLightTheme")).toBool();` such that we have one `if` with one `return`, to improve readability.

Reply:
Okay, this is now:

´´´
const auto darkModeSubkey = QStringLiteral("Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize");
    if (Utility::registryKeyExists(HKEY_CURRENT_USER, darkModeSubkey) &&
        !Utility::registryGetKeyValue(HKEY_CURRENT_USER, darkModeSubkey, QStringLiteral("AppsUseLightTheme")).toBool()) {
        return true;
    }

    return false;
´´´